aiida_quantumespresso/calculations/__init__.py

```diff
@@ -87,6 +122,35 @@ def define(cls, spec):
             help='Optional van der Waals table contained in a `SinglefileData`.')
         spec.input_namespace('pseudos', valid_type=(LegacyUpfData, UpfData), dynamic=True,
             help='A mapping of `UpfData` nodes onto the kind name to which they should apply.')
+        spec.input(
+            'parallelization',
+            valid_type=orm.Dict,
+            required=False,
+            help=(
+                'Parallelization options. The following flags are allowed:\n' + '\n'.join(
+                    f'{flag_name:<7}: {cls._PARALLELIZATION_FLAGS[flag_name]}'
+                    for flag_name in cls._ENABLED_PARALLELIZATION_FLAGS
+                )
+            ),
+            validator=cls._validate_parallelization
+        )
+
+    @classmethod
+    def _validate_parallelization(cls, value, port_namespace):  # pylint: disable=unused-argument
+        if value:
+            value_dict = value.get_dict()
+            unknown_flags = set(value_dict.keys()) - set(cls._ENABLED_PARALLELIZATION_FLAGS)
+            if unknown_flags:
+                raise exceptions.InputValidationError(
+                    f"Unknown flags in 'parallelization': {unknown_flags}, "
+                    f'allowed flags are {cls._ENABLED_PARALLELIZATION_FLAGS}.'
+                )
```

Comment:
port validators should just return an error message string, not raise

Reply:
Done!